### DISCUSSION TITLE
fastapi crashes, error: Segmentation fault: 11

### DISCUSSION THREAD

**Discussion by nafets33** [fastapi/fastapi#9715] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from chess_piece import fastapi_router
from fastapi import FastAPI, status
from fastapi.middleware.cors import CORSMiddleware
from dotenv import load_dotenv
import uvicorn

load_dotenv()


origins = []

app = FastAPI()

app.include_router(fastapi_router.router)

app.add_middleware(
    CORSMiddleware,
    allow_origins=["*"],
    allow_credentials=True,
    allow_methods=["*"],
    allow_headers=["*"],
)


@app.get("/", status_code=status.HTTP_200_OK, tags=["API Check"])
def check():
    return {
        "message": "Hello World!"
    }


if __name__ == '__main__':
    uvicorn.run(app)
```

### Description

I'm at a loss here as the fastapi server frequently crashes with error Segmentation fault: 11.

I've done my best to investigate but cannot reproduce the error, it occurs randomly. I am not sure what else to check for to help solve the issue. 

I'm hoping anyone might be able to point me in a direction on being able to debug ?

### Operating System

macOS

### FastAPI Version

fastapi==0.63.0

### Python Version

3.9.6

### Additional Context

uvicorn==0.13.3

**Comment by Kludex** [comment]
Bump `uvicorn` to the latest release may solve the issue.

**Reply by nafets33** [reply]
thanks! I upgraded hoping that solves the issue

**Reply by nafets33** [reply]
didn't work : (

**Reply by nafets33** [reply]
upgrading uvicorn==0.22.0 but still ran into same error

**Reply by Kludex** [reply]
And the FastAPI version to the latest...

**Reply by nafets33** [reply]
duhhhh yes thanks! i did and haven't seen error yet, tomorrow will tell the toll when the volume goes back up, 

I haven't seen this error when running on windows. 

fastapi is wonderful

**Reply by nafets33** [reply]
I ran into the same error: `Segmentation fault: 11` : (

**Reply by nafets33** [reply]
I found that there were a bunch of old 8000 ports sitting around that were CLOSED but still shown, so I closed them. Hoping this is the root of the error. 
I'm also investigating to see if there is any caching going on. 

I run a streamlit app and I see this when I startup the fastapi server, not sure what this is investigating...

```
pollen stefanstapinski$ python fastapi_server.py 
2023-06-27 17:37:47.191 WARNING streamlit.runtime.caching.cache_data_api: No runtime found, using MemoryCacheStorageManager
```

**Comment by nafets33** [comment]
Update >>> the issue was coming from an error reading a specific pickle file, fastapi router will read the file without issue for a couple of minutes then after ~10mins it would error out completely not being able to read the file, the issue is somewhere is fastapi, as I am able to read the file at the same time without issue. 

My current solution is that instead of reading the specific pickle file, I now break out the file into multiple files per key, and I have resolved the continuous errors. (i.e. fastapi post call now reads a different smaller file with specific contents, these smaller files contents come from the original pickle file data set)

this I believe is why I was receiving the segmentation error, fastapi gets stuck in erroring out trying to read the file and crashes. 

awesome creator of fastapi, I built a trading bot and if you have some spare time I would like to share it with you : )

**Comment by waleedrkhan90** [comment]
@nafets33 which pickle file are you referring to here ? is it a custom made file or provided by fast api ? How is it able to read line by line the  but not all file at the same time ?
i am running into the same issue we have deployed our application with kubernetes on gcp.
on my side the server crashes arent frequent but it happens. 
I am not reading any pickle file.

**Comment by nafets33** [comment, marked as answer]
My issue was that my local pkl file I was being read through the API, and
the read was failing. So my guess is the routers call was failing too many
times and FastAPI might have been caching the fail? I'm not sure but once I
solved reading my local file to catch the failure there server stopped
crashing.

~Stefan

**Reply by acidjunk** [reply]
but the example doesn't contain code that pickles?

**Reply by nafets33** [reply]
yea I didn't include the code from the router.py which reads a pickle file and returns data. regardless I don't think my issue was pickle file related, I think my issue was that my function which was trying to read a pickle file was failing.

**Reply by waleedrkhan90** [reply]
yea @nafets33  which pickle file are you talking about ? is this your own file or is this from fastapi ?

**Reply by ryukinix** [reply]
I'm experiencing a similar issue on a docker container, another related: https://stackoverflow.com/questions/77186063/python-aborted-core-dumped-while-running-fastapi-with-uvicorn-in-docker

**Comment by uxduck** [comment]
For what it's worth...I previously used Poetry, like `poetry run python main.py` to start my python app. I was living with this for months.

Switching to [uv](https://github.com/astral-sh/uv), like `uv run python main.py` fixed it.
